packages/upward-js/lib/resolvers/UrlResolver.js

```diff
@@ -0,0 +1,103 @@
+const debug = require('debug')('upward-js:UrlResolver');
+const { inspect } = require('util');
+const { URL, URLSearchParams } = require('url');
+const { forOwn, fromPairs, isPlainObject } = require('lodash');
+const AbstractResolver = require('./AbstractResolver');
+
+// The WHATWG URL object can't deal with relative URLs.
+const randomString = Math.random()
+    .toString(36)
+    .slice(2, 7);
+const FAKE_BASE_URL = new URL(`https://upward-fake-${randomString}.localhost`);
+const NON_RELATIVE_PARTS = ['protocol', 'port', 'username', 'password'];
+class UrlResolver extends AbstractResolver {
+    static get resolverType() {
+        return 'url';
+    }
+    static get telltale() {
+        return 'baseUrl';
```

Comment:
Oh, yes it can. The "telltale" concept needs to go away in UPWARD 2 anyway, but for now, it should certainly be consistent. I think the right choice is to change the implementation, not the documentation; the ServiceResolver doesn't work without `query` anyway, and no other resolver currently has a `query` argument built in. Making that change now.

Reply:
Nevermind, just ran the tests and realized that the UrlResolver does use `query` of _course_. Recommend:

- Replacing the `url` parameter to ServiceResolver with `endpoint`
- Ensuring backwards compatibility with the old `url`
- Changing the docs so that `endpoint` or the deprecated `url` is the ServiceResolver's telltale

What do you think of that?

Reply:
I like it; same verbiage that GraphiQL uses 👍

Reply:
Pushed this change.